packages/contracts-bedrock/src/dispute/OutputBisectionGame.sol

```diff
@@ -0,0 +1,551 @@
+
+import { Clone } from "src/libraries/Clone.sol";
+import { Types } from "src/libraries/Types.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+import { LibHashing } from "src/dispute/lib/LibHashing.sol";
+import { LibPosition } from "src/dispute/lib/LibPosition.sol";
+import { LibClock } from "src/dispute/lib/LibClock.sol";
+
+import "src/libraries/DisputeTypes.sol";
+import "src/libraries/DisputeErrors.sol";
+
+/// @title OutputBisectionGame
+/// @notice An implementation of the `IOutputBisectionGame` interface.
+contract OutputBisectionGame is IOutputBisectionGame, Clone, ISemver {
+    ////////////////////////////////////////////////////////////////
+    //                         State Vars                         //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice The absolute prestate of the instruction trace. This is a constant that is defined
+    ///         by the program that is being used to execute the trace.
+    Claim public immutable ABSOLUTE_PRESTATE;
+
+    /// @notice The max depth of the game.
+    uint256 public immutable MAX_GAME_DEPTH;
+
+    /// @notice The max depth of the output bisection portion of the position tree. Immediately beneath
+    ///         this depth, execution trace bisection begins.
+    uint256 public immutable SPLIT_DEPTH;
+
+    /// @notice The duration of the game.
+    Duration public immutable GAME_DURATION;
+
+    /// @notice An onchain VM that performs single instruction steps on a fault proof program trace.
+    IBigStepper public immutable VM;
+
+    /// @notice The game type ID
+    GameType internal immutable GAME_TYPE;
+
+    /// @notice The genesis block number
+    uint256 internal immutable GENESIS_BLOCK_NUMBER;
+
+    /// @notice The global root claim's position is always at gindex 1.
+    Position internal constant ROOT_POSITION = Position.wrap(1);
+
+    /// @notice The starting timestamp of the game
+    Timestamp public createdAt;
+
+    /// @notice The timestamp of the game's global resolution.
+    Timestamp public resolvedAt;
+
+    /// @inheritdoc IDisputeGame
+    GameStatus public status;
+
+    /// @inheritdoc IDisputeGame
+    IBondManager public bondManager;
+
+    /// @inheritdoc IOutputBisectionGame
+    Hash public l1Head;
+
+    /// @notice An append-only array of all claims made during the dispute game.
+    ClaimData[] public claimData;
+
+    /// @notice An internal mapping to allow for constant-time lookups of existing claims.
+    mapping(ClaimHash => bool) internal claims;
+
+    /// @notice An internal mapping of subgames rooted at a claim index to other claim indices in the subgame.
+    mapping(uint256 => uint256[]) internal subgames;
+
+    /// @notice Indicates whether the subgame rooted at the root claim has been resolved.
+    bool internal subgameAtRootResolved;
+
+    /// @notice Semantic version.
+    /// @custom:semver 0.0.13
+    string public constant version = "0.0.13";
```

Comment:
The idea here is that this is a temp contract and will delete the other contract and rename this one to the name of the other one?